core/src/main/scala/anorm/Column.scala

```diff
@@ -197,7 +197,12 @@ object Column extends JodaColumn with JavaTimeColumn {
       case long: Long => Right(long)
       case s: Short => Right(s.toLong)
       case b: Byte => Right(b.toLong)
-      case bool: Boolean => Right(if (!bool) 0l else 1l)
+      case bool: Boolean => Right(if (!bool) 0L else 1L)
+      case date: Date => Right(date.getTime)
+      case TimestampWrapper1(ts) =>
```

Comment:
I think something like this would be clearer:
```scala
case TimestampWrapper1(ts) => Right(if (ts == null) null.asInstanceOf[Long] else ts.getTime)
```

Also I think you could write a utility to share the logic among all the code blocks here dealing with dates/timestamps.